src/amber/router/parsers/multipart.cr

```diff
@@ -0,0 +1,15 @@
+module Parsers
+  module Multipart
+    def self.parse(params : Amber::Router::Params, request : HTTP::Request)
+      HTTP::FormData.parse(request) do |upload|
+        next unless upload
+        filename = upload.filename
+        if filename.is_a?(String) && !filename.empty?
+          params.files[upload.name] = Amber::Router::File.new(upload: upload)
+        else
+          params[upload.name] = upload.body.gets_to_end
```

Comment:
This needs to be `params.add(upload.name, upload.body.gets_to_end)` or else it will overwrite params that have already been defined instead of appending.